### ISSUE TITLE
Master/Detail: UISplitViewController with Storyboard (Xcode 7.1)

### ISSUE THREAD

**Issue by Souf-R** [John-Lluch/SWRevealViewController#518]
Hi,

I'm pretty new to iOS (world). I want to create a Master application in which I am using SWRevealVC as navigation drawer, currently I'm facing a problem in AppDelegate class: 

``` swift
func application(application: UIApplication, didFinishLaunchingWithOptions launchOptions: [NSObject: AnyObject]?) -> Bool {
    // Override point for customization after application launch.
    let splitViewController = self.window!.rootViewController as! UISplitViewController
    let navigationController = splitViewController.viewControllers[splitViewController.viewControllers.count-1] as! UINavigationController
    navigationController.topViewController!.navigationItem.leftBarButtonItem = splitViewController.displayModeButtonItem()
    splitViewController.delegate = self
    return true
}
```

The application crashes at the first line:

``` swift
 let splitViewController = self.window!.rootViewController as! UISplitViewController
```

The error is:

```
Could not cast value of type 'SWRevealViewController' (0xfec40) to 'UISplitViewController' (0x227256c).
```

Here is a screenshot of my **Main.storyboard**:
![capture d ecran 2015-12-01 a 11 14 03](https://cloud.githubusercontent.com/assets/7124597/11499759/5383bf1a-9820-11e5-82e4-e51ed8a2e994.png)

_PS: I'm using Xcode: 7.1, Deployment target: 8.0, , Language: Swift 2, Devices: Universal (iPhone & iPad)_

Please help me.

**Comment by bensir4u** [comment]
same here, anyone could help please.

**Comment by iDevelopper** [comment]
if you use SWRevealViewController you don't need a UISplitViewController

**Comment by Souf-R** [comment]
Thank's @iDevelopper for the reply; but could you give me a link of tutorial or a code snippet on how to use it without the UISplitViewController.
I think that I should manage this without storyboard (I mean programatically) Unfortunately, I don't know how.

Please help!

**Comment by bensir4u** [comment]
hi, @Souf-R
You may interested to take a look the link (http://www.appcoda.com/sidebar-menu-swift/)
It help you learn help to use it via storyboard, using storyboard could be easy.

**Comment by Souf-R** [comment]
Hi @bensir4u,
Thank you for the link but I already read it and I don't have a problem using SWReveal when using iPhone apps. But the deal here is that I am facing a UISplitView as mentioned in the issue description.

PS: Sorry for this late comment

Any other help!

**Comment by iDevelopper** [comment]
Keep only the iPhone storyboard and use Size class and Auto Layout

**Comment by Souf-R** [comment]
Finally, I did it programmatically, while keeping the same Storyboard.

I added some `storyboard IDs` for `master, detail, UISplit, Rear and SWReveal` views. And here is my final _didFinishLaunchingWithOptions_

``` swift
func application(application: UIApplication, didFinishLaunchingWithOptions launchOptions: [NSObject: AnyObject]?) -> Bool {
        // crete the window
        self.window = UIWindow(frame: UIScreen.mainScreen().bounds)
        self.window!.backgroundColor = UIColor.whiteColor()

        // instantiate from storyboard
        let mainStoryBoard = UIStoryboard(name: "Main", bundle: nil)
        let splitVC = mainStoryBoard.instantiateViewControllerWithIdentifier("UiSplitView") as! UISplitViewController
        let masterVC = mainStoryBoard.instantiateViewControllerWithIdentifier("masterView") as! MasterViewController
        let masterNav = UINavigationController(rootViewController: masterVC)
        let detailVC = mainStoryBoard.instantiateViewControllerWithIdentifier("detailView") as! DetailViewController
        let detailNav = UINavigationController(rootViewController: detailVC)
        // add master and detail to the SplitView
        splitVC.viewControllers = [masterNav, detailNav]

        // Override point for customization after application launch.
        let navController = splitVC.viewControllers[splitVC.viewControllers.count-1] as! UINavigationController
        navController.topViewController!.navigationItem.leftBarButtonItem = splitVC.displayModeButtonItem()
        splitVC.delegate = self

        // SWRevealViewController
        let navDrawerView = mainStoryBoard.instantiateViewControllerWithIdentifier("navDrawerView") as! NavDrawerViewController
        let swRevealView = mainStoryBoard.instantiateViewControllerWithIdentifier("swRevealView") as! SWRevealViewController
        swRevealView.setFrontViewController(splitVC, animated: true)
        swRevealView.setRearViewController(navDrawerView, animated: true)

        // Set Root view and make it visible
        self.window!.rootViewController = swRevealView
        self.window!.makeKeyAndVisible()

        return true
    }
```

I hope this will help someone else.

**Comment by thomers** [comment]
@Souf-R could you explain what you did in detail? For example, did you remove all the custom segues in the storyboard? 

I have the same setup (UISplitViewController from the Master-Detail-Xcode template, wrapped into a SWRevealViewController) and tried to translate your Swift-Code into ObjC, but only see an empty MasterViewController.

```
    self.window = [[UIWindow alloc] initWithFrame:[UIScreen mainScreen].bounds];
    self.window.backgroundColor = [UIColor whiteColor];

    UIStoryboard *mainSB = [UIStoryboard storyboardWithName:@"Main" bundle:[NSBundle mainBundle]];

    UISplitViewController *splitVC = (UISplitViewController *) [mainSB instantiateViewControllerWithIdentifier:@"SplitVC"];
    splitVC.delegate = self;

    MasterViewController *masterVC = (MasterViewController *) [mainSB instantiateViewControllerWithIdentifier:@"MasterVC"];
    UINavigationController *masterNav = [[UINavigationController alloc] initWithRootViewController:masterVC];

    DetailViewController *detailVC = (DetailViewController *) [mainSB instantiateViewControllerWithIdentifier:@"DetailVC"];
    UINavigationController *detailNav = [[UINavigationController alloc] initWithRootViewController:detailVC];

    splitVC.viewControllers = @[masterNav, detailNav];


    // SWRevealViewController
    UINavigationController *navController = (UINavigationController *) splitVC.viewControllers[splitVC.viewControllers.count-1];
    navController.topViewController.navigationItem.leftBarButtonItem = splitVC.displayModeButtonItem;
    splitVC.delegate = self;

    SWRevealViewController *swRevealVC = [mainSB instantiateViewControllerWithIdentifier:@"SWRevealVC"];
    UIViewController *menuVC = [mainSB instantiateViewControllerWithIdentifier:@"MenuVC"];
    [swRevealVC setFrontViewController:splitVC animated:YES];
    [swRevealVC setRearViewController:menuVC animated:YES];

    self.window.rootViewController = swRevealVC;
    [self.window makeKeyAndVisible];

```

**Comment by Souf-R** [comment]
Hi @thomers,

1- I created the masterDetail project using XCode Assistant (wizard)
2- I modified the main storyboard as mentioned in my first post (See the image in the question)
3- I added the SWRevealViewController (See the image) then the segue IDs (names)
4- I modified my `application:didFinishLaunchingWithOptions` function

And that's it.

**Comment by Souf-R** [comment]
Hi @thomers,

1- I created the masterDetail project using XCode Assistant (wizard)
2- I modified the main storyboard as mentioned in my first post (See the image in the question)
3- I added the SWRevealViewController (See the image) then the segue IDs (names)
4- I modified my `application:didFinishLaunchingWithOptions` function

And that's it.

**Comment by thomers** [comment]
Thanks for the reply. What I was missing was the correct segue type of "reveal view controller set segue" for the "sw_front" and "sw_rear" segues in the storyboard.

**Comment by thomers** [comment]
One more thing - does the menu open correctly for you on iPad in Portrait orientation?

I have added the "Menu" BarButtonItem to the MasterViewController, and call "revealToggle".

When I only show the DetailViewController, the left BarButtonItem shows "Master". I tap "Master", the master Is shown on the left side, overlapping the DetailViewController correctly. But when I now tap "Menu" in the MasterViewController, the MenuViewController is "shown" (actually hidden) underneath the MasterViewController, and shifts the DetailViewController to the right.

tl;dr: The "sw_rear" ViewController is hidden on iPad in Portrait orientation.

**Comment by thomers** [comment]
And another one: 

I can show a different ViewController via a _SWRevealViewControllerSeguePushController_ segue from the MenuTableViewController. This works.

But when I then want to show the UISplitViewController again, the DetailViewController is shown instead of the MasterViewController, and I have to tap "Master" to show the MasterViewController.

What do I have to do to show the MasterDetailViewController (like it works after the first app launch)?

<img width="1657" alt="screen shot 2016-03-09 at 15 56 58" src="https://cloud.githubusercontent.com/assets/453568/13639337/986db34a-e610-11e5-973f-0ff530193ecc.png">

**Comment by thomers** [comment]
(Hmm, it's probably better to open a new issue)

**Comment by thomers** [comment]
Please see #559
